wagtail/admin/templatetags/wagtailadmin_tags.py

```diff
@@ -872,6 +874,22 @@ def preview_settings():
     }
 
 
+@register.simple_tag(takes_context=True)
+def wagtail_config(context):
+    request = context["request"]
+    config = {
+        "CSRF_TOKEN": get_token(request),
+        "CSRF_HEADER_NAME": HttpHeaders.parse_header_name(
```

Comment:
Django can't handle CSRF tokens in the body if it's not a POST request, so we have to pass it in the header. Meanwhile, the header name can be changed using Django's `CSRF_HEADER_NAME`, so we need to pass this to the JS as well.